src/botPage/view/blockly/blocks/trade/tradeOptions.js

```diff
@@ -199,7 +199,8 @@ export default () => {
                     const barrierLabels = [translate('High barrier'), translate('Low barrier')];
                     const removeInput = inputName => tradeOptionsBlock.removeInput(inputName);
 
-                    const updateList = (list, options) => {
+                    const updateList = (list, options = []) => {
+                        if (!options?.length) return;
                         const prevSelectedType = options.find(option => option[1] === list?.getValue());
                         list.menuGenerator_ = options; // eslint-disable-line no-underscore-dangle, no-param-reassign
```

Comment:
I think the problem is not connected with options. 
You've get a error with message "Cannot set properties of null (setting 'menuGenerator_')"
So it means that something before .menuGenerator_ could be null. and it causes an error. 
To fix it i guess you should do something like this, only on line: 205
` list?.menuGenerator_ = options`

this line means something like 
```
if (!list) {
 list.menuGenerator_ = options;
}

```

Reply:
oh, you're right Hena. "list" is unavailable here, not "options". But instead of adding optional chaining in one place we need to check it at very top. Because then the error can potentially occur at further "list" methods calls.

Reply:
Made suggested changes, and I believe this is correct. Kindly review again.